upgrade/upgrade-scripts/execute

```diff
@@ -459,6 +478,42 @@ else
 			"for '$ROOTFS_CONTAINER'"
 fi
 
+# In AWS environments, changing instance type could result in generation of
+# different network interface names compared to the existing ones. This mismatch
+# causes our netplan file handler in the osadmin service to remove the existing
+# entries in the netplan plan, because they are not to be found on the new instance.
+# However, the mac address remains consistent and does not change with a change in
+# the instance types.
+# To handle this, from 6.0.16.0 onwards, we are switching to use mac address based
+# interface names in the netplan file. To facilitate this change across upgrades from
+# older verrsions to 6.0.16.0 and above, we create a map from the network interface
+# names to their corresponding mac address, resulting in mac addrress based interface
+# names. For wider context, see DLPX-75209.
+#
+# For deferred upgrades, we cannot know when the engine will be rebooted into the new
+# image. The decision depends on the end-user. Hence we always create the map file
+# irrespective of the upgrade type, to allow the upgrade finalize logic to copy the
+# map file from the currently mounted root filesystem to the root filesystem of the
+# target container.
+#
+# The map creation logic does not apply to engines being upgraded from 6.0.16.0 onwards.
+# This is because the netplan file from 6.0.16.0 will by default be populated with mac
+# address based interface names. This makes this logic redundant for upgrades from
+# 6.0.16.0 and onwards. Hence, we must remove this logic as soon as we mark 6.0.16.0 as
+# the minimum version supported release.
+
+# The container does not have visibility into the complete network namespace of the
+# engine. Hence we need to create the map outside the container to ensure the presense
+# of all the interfaces in the map
+if ! systemd-detect-virt --container --quiet && [[ $(get-appliance-platform) == "aws" ]]; then
```

Comment:
will the `generate_interface_to_mac_address_map` actually fail if it's run in the container? just curious if we need to have it be conditional or not.. this check will only prevent us calling the function in the container.. it'll run outside the container no matter what..

Reply:
In my tests, I was never able to get this map generated from inside the container because the container had no visibility into the net namespace.  In the implementation of `generate_interface_to_mac_address_map` function, we call the die() function at the end if the generated map file is empty. 
It is probably not a bad idea to make it explicit that this function should not be called in the container ? Let me know if you feel otherwise.

Reply:
ah, ok.. so it's a self-inflicted failure, since the file is empty.. I'm OK with that, just was curious.. we can keep it as-is.